partiql-logical-planner/src/typer.rs

```diff
@@ -0,0 +1,757 @@
+                                        new_type_env.insert(b.0, b.1);
+                                    });
+                                } else {
+                                    new_type_env.insert(sym, ty.clone());
+                                }
+
+                                let derived_type = self.derived_type(&type_ctx);
+                                let new_ty = self.element_type(&derived_type);
+                                let type_ctx = ty_ctx![(&new_type_env, &new_ty)];
+                                self.type_env_stack.push(type_ctx);
+                                break;
+                            }
+                        }
+                    }
+                }
+            }
+            ValueExpr::Path(v, components) => {
+                self.type_vexpr(v, LookupOrder::LocalGlobal);
+                // let type_ctx = self.local_type_env();
+                for component in components {
+                    match component {
+                        PathComponent::Key(key) => {
+                            let var = ValueExpr::VarRef(key.clone());
+                            self.type_vexpr(&var, LookupOrder::LocalGlobal);
+                        }
+                        PathComponent::Index(_) => {
+                            self.errors.push(TypingError::NotYetImplemented(
+                                "Typing [Index] [PathComponent]s".to_string(),
+                            ))
+                        }
+                        PathComponent::KeyExpr(_) => {
+                            self.errors.push(TypingError::NotYetImplemented(
+                                "Typing [KeyExpr] [PathComponent]s".to_string(),
+                            ))
+                        }
+                        PathComponent::IndexExpr(_) => {
+                            self.errors.push(TypingError::NotYetImplemented(
+                                "Typing [IndexExpr] [PathComponent]s".to_string(),
+                            ))
+                        }
+                    }
+                }
+            }
+            ValueExpr::Lit(v) => {
+                let kind = match **v {
+                    Value::Null => TypeKind::Null,
+                    Value::Missing => TypeKind::Missing,
+                    Value::Integer(_) => TypeKind::Int,
+                    Value::Decimal(_) => TypeKind::Decimal,
+                    Value::Boolean(_) => TypeKind::Bool,
+                    Value::String(_) => TypeKind::String,
+                    Value::Tuple(_) => TypeKind::Struct(StructType::new_any()),
+                    Value::List(_) => TypeKind::Array(ArrayType::new_any()),
+                    Value::Bag(_) => TypeKind::Bag(BagType::new_any()),
+                    _ => {
+                        self.errors.push(TypingError::NotYetImplemented(
+                            "Unsupported Literal".to_string(),
+                        ));
+                        todo!()
```

Comment:
What's the todo for?

Reply:
This should not be here, will clean it up.